tests/test_main.py

```diff
@@ -0,0 +1,209 @@
+import os
+import random
+import string
+from importlib import reload
+from unittest import mock
+from unittest.mock import Mock
+from uuid import uuid4
+
+import pytest
+import responses
+
+import main
+from exceptions import InvalidNotifyKeyError
+from main import NOTIFY_BASE_URL, send_email
+
+url = f"{NOTIFY_BASE_URL}/notifications/email"
+
+
+def test_get_not_allowed():
+    request = Mock(method="GET")
+    response = send_email(request)
+    assert response == ("Method not allowed", 405)
+
+
+def test_missing_data_returns_422():
+    request = Mock(method="POST", json={})
+    response = send_email(request)
+    assert response == ("Missing notification request data", 422)
+
+
+def test_invalid_service_id_raises_invalid_notify_key_error():
+    random_string = "".join(random.choice(string.printable) for i in range(87))
+    with mock.patch.dict(os.environ, {"NOTIFY_API_KEY": random_string}):
+        with pytest.raises(InvalidNotifyKeyError):
+            reload(main)
+
+
+def test_invalid_api_key_raises_invalid_notify_key_error():
+    random_string = "".join(random.choice(string.printable) for i in range(37))
+    uuid_string = str(uuid4())
+    with mock.patch.dict(os.environ, {"NOTIFY_API_KEY": uuid_string + random_string}):
+        with pytest.raises(InvalidNotifyKeyError):
+            reload(main)
+
+
+@responses.activate
+def test_notify_response_error_returns_correctly():
+    request = Mock(
+        method="POST",
+        json={
+            "payload": {
+                "fulfilmentRequest": {
+                    "email_address": "test@example.com",
+                    "personalisation": {"address": "test address"},
+                    "form_type": "H",
+                    "language_code": "en",
+                    "region_code": "GB-ENG",
+                }
+            },
+        },
+    )
```

Comment:
This block is repeated in a few tests and could be factored out. It would also make it clearer that these don't vary between tests and are just repeated valid requests.

I think these also need to be updated to use `display_address`.

Reply:
Added a fixture for this, and added the `display_address` property